ward/testing.py

```diff
@@ -104,12 +105,62 @@ class Test:
     sout: StringIO = field(default_factory=StringIO)
     serr: StringIO = field(default_factory=StringIO)
     ward_meta: WardMeta = field(default_factory=WardMeta)
+    timer: Optional["Timer"] = None
+    args: Optional[Dict[str, Any]] = None
 
-    def __call__(self, *args, **kwargs):
-        if self.capture_output:
-            with redirect_stdout(self.sout), redirect_stderr(self.serr):
-                return self.fn(*args, **kwargs)
-        return self.fn(*args, **kwargs)
+    def run(self, cache: FixtureCache, idx: int = 0) -> "TestResult":
+
+        with ExitStack() as stack:
+            self.timer = stack.enter_context(Timer())
+            if self.capture_output:
+                stack.enter_context(redirect_stdout(self.sout))
+                stack.enter_context(redirect_stderr(self.serr))
+
+            if isinstance(self.marker, SkipMarker):
+                # TODO:onlyanegg: Do sout and serr need to be part of the class? If they weren't we could have
+                # the ExitStack close them
+                with closing(self.sout), closing(self.serr):
+                    result = TestResult(self, TestOutcome.SKIP)
+                return result
+
+            try:
+                # TODO:onlyanegg: I don't love this. We're setting up the
+                # fixture within the testing module, but cleaning it up in the
+                # suite module.
+                self.resolve_args(cache, iteration=idx)
+                self.format_description()
```

Comment:
I think it'd be better to return the dict from `resolved_args` (rather than setting `self.args`) and pass that into `format_description`, otherwise there's a hidden coupling between the methods. If the dict is returned, it implies we can't use `format_description` without it.

Reply:
That makes a lot of sense.